src/core/server/audit_trail/types.ts

```diff
@@ -19,43 +19,120 @@
 import { KibanaRequest } from '../http';
 
 /**
- * Event to audit.
+ * Audit event schema using ECS format.
+ * https://www.elastic.co/guide/en/ecs/1.5/index.html
  * @public
- *
- * @remarks
- * Not a complete interface.
  */
-export interface AuditableEvent {
+export interface AuditEvent {
+  /**
+   * Human readable message describing the action, outcome and user.
+   *
+   * @example
+   * User 'jdoe' updated saved object 'kpis' of type 'dashboard'
+   */
   message: string;
-  type: string;
+  event: {
+    action: string;
+    category: EventCategory | readonly EventCategory[];
+    type: EventType | readonly EventType[];
+    outcome: EventOutcome;
+    module?: string;
+    dataset?: string;
+  };
+  user: {
+    name: string;
+    email?: string;
+    full_name?: string;
+    hash?: string;
+    roles?: readonly string[];
+  };
+  session?: {
+    id: string;
+  };
+  kibana: {
+    /**
+     * Current space id of the request.
+     */
+    namespace: string;
+    /**
+     * Array of saved objects created, accessed, updated or deleted as part of the action.
+     */
+    saved_objects?: ReadonlyArray<{
+      type: string;
+      id?: string;
+      /**
+       * Array of namespaces added to or removed from the saved object.
+       */
+      namespaces?: string[];
+    }>;
+  };
+  error?: {
+    code?: string;
+    message?: string;
+  };
+  http?: {
+    request?: {
+      method?: string;
+      body?: {
+        content: string;
+      };
+    };
+    response?: {
+      status_code?: number;
```

Comment:
Would we have a `http.response` defined without a `http.response.status_code`? Would this be more accurate?

```suggestion
      status_code: number;
```